x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -6722,4 +6722,26 @@ public void testMatchFunctionIsNotNullable() {
             containsString("[MATCH] function cannot operate on [text::keyword], which is not a field from an index mapping")
         );
     }
+
+    public void testWhereNull() {
+        var plan = plan("""
+            from test
+            | sort salary
+            | rename emp_no as e, first_name as f
+            | keep salary, e, f
+            | where null
+            | LIMIT 12
+            """);
+        var local = as(plan, LocalRelation.class);
+        assertThat(local.supplier(), equalTo(LocalSupplier.EMPTY));
+
+        VerificationException e = expectThrows(VerificationException.class, () -> plan("""
+            from test
+            | where concat(first_name, "foobar")
+            """));
+        assertThat(
+            e.getMessage(),
+            containsString("Found 1 problem\nline 2:9: Condition expression needs to be boolean, found [KEYWORD]")
+        );
```

Comment:
This test can be moved to `VerifierTests`, as the `VerificationException` is thrown before `LogicalPlanOptimizer`.